### DISCUSSION TITLE
Is it possible to add a new metric

### DISCUSSION THREAD

**Discussion by sandraplato1234** [milvus-io/milvus#4692] [Q&A and General discussion]
Hi,

I need to add a statistical metric like kendall.
Is it possible?
Thanks in advance

**Comment by shengjun1985** [comment]
Our data is stored in segments, but the statistical information refers to the full amount of data.
So in the current architecture, the feature is hard to support.
But your requirement is very interesting, we will consider it in our new designs.
By the way, if we split the data into segments, and then we only count the data distribution within each segment. Can you accept it?

**Comment by gujun720** [comment]
Is it [Kendall tau distance](https://en.wikipedia.org/wiki/Kendall_tau_distance)?
Can you describe the application scenario and the potential data volume?

**Reply by sandraplato1234** [reply]
Yes it kendall tau distance.
The scenario is image vectors to compare
in the collection there are ~ 10000 image vectors

**Comment by shengjun1985** [comment]
@sandraplato1234
It's not a KNN problem. It's a statistic.
What you expect is an interface to get the Kendall coefficients of some dimensions in a collection, isn't it?
What dimensions do you use now?

**Reply by sandraplato1234** [reply]
Hi
The dimensions is 1408 
I'm not sure I understand your question about the interface I expected ,I expected that the metric will be kendall instead of IP and use the search method in the api

**Reply by shengjun1985** [reply]
@sandraplato1234 
I got it. Thank you.
Can you write a C + + program? If you can, would you like to write a calculation function of Kendall in [distance.cpp](https://github.com/milvus-io/milvus/blob/master/core/src/index/thirdparty/faiss/utils/distances.cpp)? Then, we can implement a flat search first.
In addition, does Kendall satisfy trigonometric inequality (|AB| + |AC| >= |BC|)? If it satisfies, we can try to index it later.
